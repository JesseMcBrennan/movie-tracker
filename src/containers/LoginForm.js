import React, { Component } from 'react';
import { connect } from 'react-redux';
import { getUserInfo, populateFavorites } from '../actions';
import { withRouter } from 'react-router-dom';
import { fetchUserData, getUserFavorites } from '../utils/apiCalls';
import PropTypes from 'prop-types';

class LoginForm extends Component {
  constructor() {
    super();
    this.state = {
      email: '',
      password: ''
    };
  }

  handleChange = e => {
    const { name, value } = e.target;
    this.setState({
      [name]: value
    });
  };

  handleSubmit = async e => {
    e.preventDefault();
    const result = await fetchUserData(this.state.email, this.state.password);
    if (result) {
      const { name, id } = result.data;
      this.props.userLogin(name, id);
      const userFavorites = await getUserFavorites(id);
      const userFavoritesIds = userFavorites.map(favorite => favorite.movie_id);
      this.props.getFavorites(userFavoritesIds);
      this.props.history.push('/');
    }
    return;
  };

  render() {
    return (
      <form onSubmit={this.handleSubmit}>
        <input
          name="email"
          value={this.state.email}
          onChange={this.handleChange}
          placeholder="Email"
          type="text"
        />
        <input
          name="password"
          value={this.state.password}
          onChange={this.handleChange}
          placeholder="Password"
          type="text"
        />
        <button>LOGIN</button>
      </form>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  userLogin: (name, id) => dispatch(getUserInfo(name, id)),
  getFavorites: favorites => dispatch(populateFavorites(favorites))
});

export default withRouter(
  connect(
    null,
    mapDispatchToProps
  )(LoginForm)
);

LoginForm.propTypes = {
  userLogin: PropTypes.func,
  getFavorites: PropTypes.func
};
