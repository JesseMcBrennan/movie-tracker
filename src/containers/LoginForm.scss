form {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 1rem;
}

div {
  text-align: center;
}

input {
  font-size: 1rem;
  height: 22px;
  margin-bottom: 0.9rem;
  width: 250px;
}

::placeholder {
  font-size: 1rem;
}

.login-image {
  width: 250px;
  min-height: 2em;
  border: none;
  box-shadow: grey;
  padding: 1rem 0 0 0rem;
}

.login-button {
  border-radius: 0px;
  height: 30px;
  width: 257px;
}

.login-section {
  display: flex;
  justify-content: space-evenly;
  padding-left: 33.4rem;
}

.login-home-link {
  padding: 1rem 0 0 27rem;
}
