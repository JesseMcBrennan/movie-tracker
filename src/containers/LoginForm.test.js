import { LoginForm, mapStateToProps, mapDispatchToProps } from './LoginForm'

describe('LoginForm', () => {
  
})