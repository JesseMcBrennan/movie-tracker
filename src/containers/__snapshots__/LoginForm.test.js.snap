// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginForm
    userLogin={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <section
          className="login-section"
>
          <img
                    alt="netflix-font"
                    border="0"
                    className="login-image"
                    src="https://fontmeme.com/permalink/180902/ac7a65b3a8454e0ff8d71d4efa88c41c.png"
          />
          <NavLink
                    activeClassName="active"
                    aria-current="page"
                    className="login-home-link"
                    to="/"
          >
                    Home
          </NavLink>
</section>,
        <form
          onSubmit={[Function]}
>
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email"
                    type="text"
                    value=""
          />
          <input
                    name="password"
                    onChange={[Function]}
                    placeholder="Password"
                    type="text"
                    value=""
          />
          <button
                    className="login-button"
          >
                    LOGIN
          </button>
</form>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="netflix-font"
              border="0"
              className="login-image"
              src="https://fontmeme.com/permalink/180902/ac7a65b3a8454e0ff8d71d4efa88c41c.png"
/>,
            <NavLink
              activeClassName="active"
              aria-current="page"
              className="login-home-link"
              to="/"
>
              Home
</NavLink>,
          ],
          "className": "login-section",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "netflix-font",
              "border": "0",
              "className": "login-image",
              "src": "https://fontmeme.com/permalink/180902/ac7a65b3a8454e0ff8d71d4efa88c41c.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "aria-current": "page",
              "children": "Home",
              "className": "login-home-link",
              "to": "/",
            },
            "ref": null,
            "rendered": "Home",
            "type": [Function],
          },
        ],
        "type": "section",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email"
              type="text"
              value=""
/>,
            <input
              name="password"
              onChange={[Function]}
              placeholder="Password"
              type="text"
              value=""
/>,
            <button
              className="login-button"
>
              LOGIN
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "password",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "LOGIN",
              "className": "login-button",
            },
            "ref": null,
            "rendered": "LOGIN",
            "type": "button",
          },
        ],
        "type": "form",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <section
            className="login-section"
>
            <img
                        alt="netflix-font"
                        border="0"
                        className="login-image"
                        src="https://fontmeme.com/permalink/180902/ac7a65b3a8454e0ff8d71d4efa88c41c.png"
            />
            <NavLink
                        activeClassName="active"
                        aria-current="page"
                        className="login-home-link"
                        to="/"
            >
                        Home
            </NavLink>
</section>,
          <form
            onSubmit={[Function]}
>
            <input
                        name="email"
                        onChange={[Function]}
                        placeholder="Email"
                        type="text"
                        value=""
            />
            <input
                        name="password"
                        onChange={[Function]}
                        placeholder="Password"
                        type="text"
                        value=""
            />
            <button
                        className="login-button"
            >
                        LOGIN
            </button>
</form>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="netflix-font"
                border="0"
                className="login-image"
                src="https://fontmeme.com/permalink/180902/ac7a65b3a8454e0ff8d71d4efa88c41c.png"
/>,

              <NavLink
                activeClassName="active"
                aria-current="page"
                className="login-home-link"
                to="/"
>
                Home
</NavLink>,
            ],
            "className": "login-section",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "netflix-font",
                "border": "0",
                "className": "login-image",
                "src": "https://fontmeme.com/permalink/180902/ac7a65b3a8454e0ff8d71d4efa88c41c.png",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,

              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Home",
                "className": "login-home-link",
                "to": "/",
              },
              "ref": null,
              "rendered": "Home",
              "type": [Function],
            },
          ],
          "type": "section",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                name="email"
                onChange={[Function]}
                placeholder="Email"
                type="text"
                value=""
/>,
              <input
                name="password"
                onChange={[Function]}
                placeholder="Password"
                type="text"
                value=""
/>,
              <button
                className="login-button"
>
                LOGIN
</button>,
            ],
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "email",
                "onChange": [Function],
                "placeholder": "Email",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "name": "password",
                "onChange": [Function],
                "placeholder": "Password",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "LOGIN",
                "className": "login-button",
              },
              "ref": null,
              "rendered": "LOGIN",
              "type": "button",
            },
          ],
          "type": "form",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
